src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -386,10 +540,26 @@ private boolean ignoreEvent(CommentAdded event) {
      * @param join if the method should wait for the thread to finish before returning.
      */
     public void shutdown(boolean join) {
-        for (EventThread worker : workers) {
-            worker.shutdown();
+      if (executor != null) {
+        ThreadPoolExecutor pool = executor;
+        executor = null;
+        pool.shutdown(); // Disable new tasks from being submitted
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!pool.awaitTermination(WAIT_FOR_JOBS_SHUTDOWN_TIMEOUT, TimeUnit.SECONDS)) {
```

Comment:
You should only `awaitTermination` if `join == true`

Reply:
done